### ISSUE TITLE
S101: Configuration for custom acronyms

### ISSUE THREAD

**Issue by dnperfors** [SonarSource/sonar-dotnet#2440]
### Description

On our project we have some three and four letter acronyms, that are part of a specification. These acronyms are used in code, but they will trigger S101.
Would it be possible to add a file to the project that describe what acronyms are allowed, or should be written, for example in a similar way as the CustomDictionary.xml file that is used by [CA1709](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1709-identifiers-should-be-cased-correctly?view=vs-2019)

### Repro steps

Create a class with the name `NSAInformer`

### Expected behavior

Warning S101 is not given for NSAInformer, because NSA is the established capitalized 3 letter acronym in our specification.

### Actual behavior

Warning S101 is given NSAInfomer, telling me that it should be NsaInformer

### Known workarounds

Suppress S101 in all cases where the NSA acronym is used

### Related information

* SonarAnalyzer.CSharp version 7.14.0.8411
* Visual Studio Version 2019 (16.1.1)

**Comment by andrei-epure-sonarsource** [comment]
RSPEC-101

SonarWay
Default Severity:Minor
Impact:Low
Likelihood:Low

**Comment by MichelZ** [comment]
Yeah, this one feels annoying. Ideally it should take the resharper .DotSettings and the VS equivalent to ignore certain abbreviations.
Alternatively use your own dictionary, that could be generated then from existing files by the community. As it stands now, we will also just ignore S101 :(

**Comment by ericregnier** [comment]
@andrei-epure-sonarsource any updates? :)

**Comment by HaGGi13** [comment]
@andrei-epure-sonarsource 
Hello, any updates? It would be great to have something like this. Otherwise we start to ignore [S101](https://sonarsource.github.io/rspec/#/rspec/S101/csharp) as well. And I'm really not in favor with this approach.

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1990](https://sonarsource.atlassian.net/browse/NET-1990)

[NET-1990]: https://sonarsource.atlassian.net/browse/NET-1990?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by kimpenhaus** [comment]
@pavel-mikula-sonarsource i was following this issue but don't have access to your linked jira work item. is this ticket solved in a way that acronyms are now supported (as your are closing this request)? how could one get acces to the jira ticket?

<img width="200" alt="Image" src="https://github.com/user-attachments/assets/0d699c6d-ef03-4ffb-ab40-87612a203e2e" />

thanks, m.

**Comment by pavel-mikula-sonarsource** [comment]
I'll reopen this one so you get an update here once it's done. We've generally left open mainly FPs and moved most of the tooling/improvement ones

**Comment by Webreaper** [comment]
Is there any possibility to get an ETA for this? The Jira was opened > 6mo ago, and it's impossible to know whether it's coming in an imminent release, or hasn't been worked on at all.

**Comment by pavel-mikula-sonarsource** [comment]
I've bumped the priority by one notch, but we can't provide any ETA unfortunately.

Once it makes it to a release, we'll close this ticket.
